src/mysql_k8s_helpers.py

```diff
@@ -335,16 +409,221 @@ def execute_backup_commands(
+
+        try:
+            process = self.container.exec(
+                restore_backup_command,
+                user="mysql",
+                group="mysql",
+            )
+            stdout, stderr = process.wait_output()
+            return (stdout, stderr)
+        except ExecError as e:
+            logger.exception("Failed to restore backup", exc_info=e)
+            logger.error(f"Stdout: {e.stdout}")
+            logger.error(f"Stderr: {e.stderr}")
+            raise MySQLRestoreBackupError(e.stderr)
+
+    def _get_total_memory(self) -> int:
+        """Retrieves the total memory of the mysql container."""
+        try:
+            logger.info("Retrieving the total memory of the mysql container")
+
+            """Below is an example output of `free --bytes`:
+               total        used        free      shared  buff/cache   available
+Mem:     16484458496 11890454528   265670656  2906722304  4328333312  1321193472
+Swap:     1027600384  1027600384           0
+            """
+            # need to use sh -c to be able to use pipes
+            get_total_memory_command = [
+                "sh",
+                "-c",
+                "free --bytes | sed -n '2p' | awk '{print $2}'",
+            ]
+
+            process = self.container.exec(
+                get_total_memory_command,
+                user="mysql",
+                group="mysql",
+            )
+            stdout, _ = process.wait_output()
+            return int(stdout.strip())
+        except ExecError as e:
+            logger.exception("Failed to execute commands to query total memory", exc_info=e)
+            raise
+
+    def get_innodb_buffer_pool_parameters(self) -> Tuple[int, Optional[int]]:
+        """Get innodb buffer pool parameters for the instance.
+
+        Returns: a tuple of (innodb_buffer_pool_size, optional(innodb_buffer_pool_chunk_size))
+        """
+        # Reference: based off xtradb-cluster-operator
+        # https://github.com/percona/percona-xtradb-cluster-operator/blob/main/pkg/pxc/app/config/autotune.go#L31-L54
+
+        chunk_size_min = 1048576  # 1 mebibyte
+        chunk_size_default = 134217728  # 128 mebibytes
+
+        try:
+            innodb_buffer_pool_chunk_size = None
+            total_memory = self._get_total_memory()
+
+            pool_size = int(total_memory * 0.75)
+            # 1000000000 = 1 gigabyte
+            if total_memory - pool_size < 1000000000:
+                pool_size = int(total_memory * 0.5)
+
+            if pool_size % chunk_size_default != 0:
+                # round pool_size to be a multiple of chunk_size_default
+                pool_size += chunk_size_default - (pool_size % chunk_size_default)
+
+            # 1073741824 = 1 gibibyte
+            if pool_size > 1073741824:
+                chunk_size = pool_size / 8
+                # round chunk_size to a multiple of chunk_size_min
+                chunk_size = chunk_size + chunk_size_min - (chunk_size % chunk_size_min)
```

Comment:
I would do this instead
```suggestion
                if chunk_size % chunk_size_min != 0:
                    chunk_size += chunk_size_min - (chunk_size % chunk_size_min)
```

Reply:
addressed in f03e92b4e45ec051635ce73a25ba64629e0eaa03